src/built_ins.md

```diff
@@ -471,204 +493,266 @@ push(Array<float64>, float64): Array<float64>
+set(Array<int64>, int64, int64): Result<Array<int64>>
+set(Array<float32>, int64, float32): Result<Array<float32>>
+set(Array<float64>, int64, float64): Result<Array<float64>>
+set(Array<bool>, int64, bool): Result<Array<bool>>
+```
+
 ##### HashMap Manipulation
 
-```rust
+```rust,ignore
+// Converts a value into a hash using the [xxh64](https://cyan4973.github.io/xxHash/) algorithm.
+// No user code ever has to overload the `toHash` function with their own implementation, but it is
+// required that the `eq` function is for user-defined types to check for hash collisions.
+toHash(any): int64
+toHash(int8): int64
+toHash(int16): int64
+toHash(int32): int64
+toHash(float32): int64
+toHash(float64): int64
+toHash(bool): int64
+```
+
+```rust,ignore
 // Takes a HashMap and returns an Array of KeyVal pairs
 keyVal(HashMap<Hashable, any>): Array<KeyVal<Hashable, any>>
 ```
 
-```rust
+```rust,ignore
 // Takes a HashMap and returns an Array of keys
 keys(HashMap<Hashable, any>): Array<Hashable>
 ```
 
-```rust
+```rust,ignore
 // Takes a HashMap and returns an Array of values
 values(HashMap<Hashable, any>: Array<any>
 ```
 
-```rust
+```rust,ignore
 // Takes a HashMap and returns the number of KeyVal pairs contained
 length(HashMap<Hashable, any>): int64
 ```
 
+```rust,ignore
+// Sets a key-value pair in the HashMap and returns the HashMap for chaining
+set(HashMap<Hashable, any>, Hashable, any): HashMap<Hashable, any>
+```
+
+```rust,ignore
+// Gets the value for a given key or an error
+get(HashMap<Hashable,any>, Hashable): Result<any>
+```
+
+```rust,ignore
+// Creates a new HashMap object with the first key-val pair and a decently-sized set of defaults for
+// the indexing
+newHashMap(Hashable, any): HashMap<Hashable, any>
+```
+
+```rust,ignore
+// The inverse of `keyVal`, takes an `Array<KeyVal<Hashable, any>>` and reindexes it as HashMap
+toHashMap(Array<KeyVal<Hashable, any>>): HashMap<Hashable, any>
+```
+
 ##### "Ternary" Functions
 
-```rust
+```rust,ignore
 // Takes two values of the same type and returns an array of those two values
 pair(any, any): Array<any>
 ```
 
-```rust
+```rust,ignore
 // Takes a boolean and an array of two values and returns the first value on `true` and the second
 // on `false`
 cond(bool, Array<any>): any
 ```
 
-```rust
+```rust,ignore
 // Takes a boolean and a function and conditionally executes that function if the bool is `true`
 cond(bool, function): void
 ```
 
-##### Assign Functions
+##### Clone Functions
 
-```rust
-// Explicitly duplicates the provided value and returns it
-assign(any): any
-assign(Array<any>): Array<any>
-assign(void): void
-assign(int8): int8
-assign(int16): int16
-assign(int32): int32
-assign(int64): int64
-assign(float32): float32
-assign(float64): float64
-assign(bool): bool
-assign(string): string
+```rust,ignore
+// Explicitly duplicates the provided value and returns it, useful when you don't want to mutate
+// the original value
```

Comment:
did we forget to add `HashMap` or is not here on purpose?

Reply:
It's caught by the `clone(any): any` implementation since `HashMap` is just a regular user-defined type, so it didn't need any special treatment.